lib/json/ld/frame.rb

```diff
@@ -490,7 +500,11 @@ def add_frame_output(parent, property, output)
     # @param [Hash] flags the current framing flags.
     # @return [Array<Hash>] the implicit frame.
     def create_implicit_frame(flags)
-      flags.keys.inject({}) {|memo, key| memo["@#{key}"] = [flags[key]]; memo}
+      {}.tap do |memo|
+        flags.each_pair do |key, val|
+          memo["@#{key}"] = [val]
+        end
+      end
     end
```

Comment:
Another interesting idiom.

Reply:
I used `{}.tap` to emulate the following:

```ruby
flags.each_with_object({}) do |(key, val), memo|
  memo["@#{key}"] = [val]
end
```
but without having a [key,val] array instantiated for technical reasons.

Using stackprof in object mode is great to know the impact in term of object allocation of every line in the code.